### ISSUE TITLE
BUG: Self-damaging "buffs" seem to apply the damage animation only once

### ISSUE THREAD

**Issue by jadebenn** [DarkflameUniverse/DarkflameServer#1284]
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### DarkflameServer Version

411dce7

### Issue Description

I previously noted this as a code comment in pull request #1276, but I'm making this into an actual issue for easier tracking. While taking damage in general can sometimes be inconsistent as to whether the animation is applied (I may make a separate issue for that), it's much easier to test this particular damage animation issue - even if I still don't understand what's causing it.

https://github.com/DarkflameUniverse/DarkflameServer/assets/9892985/f658c27a-22ee-408b-a615-051d3f2ae339

### Reproduction Steps

1. Pull out a Maelstrom Dagger
2. Hold it
3. Observe that health decrements without the corresponding damage animation after the first time

### Expected Behavior

The damage animation should play when the player character takes damage

### Environment

Windows client, WSL Ubuntu server

**Comment by jadebenn** [comment]
To add to this: When I was testing all the damage cooldown code and noticed this, I temporarily added a log entry right above the "onhit" animation statement and compiled the server. The log would display in the server console with every damage tick. The animation would not.

**Comment by EmosewaMC** [comment]
This is the correct behavior, the issue is that the effect doesnt play.
